Helper/ManagementHelper.php

```diff
@@ -40,35 +43,86 @@ class ManagementHelper
      * @var StoreManager
      */
     private $storeManager;
+    /**
+     * @var Config
+     */
+    private $configHelper;
 
     /**
      * ManagementHelper constructor.
      * @param StoreManager $storeManager
      * @param Data $adyenHelper
-     * @throws \Adyen\AdyenException
-     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @param Config $configHelper
      */
-    public function __construct(StoreManager $storeManager, Data $adyenHelper)
+    public function __construct(StoreManager $storeManager, Data $adyenHelper, Config $configHelper)
     {
         $this->adyenHelper = $adyenHelper;
         $this->storeManager = $storeManager;
+        $this->configHelper = $configHelper;
     }
 
     /**
      * @param string $xapikey
      * @return array
-     * @throws \Adyen\AdyenException
-     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws AdyenException
+     * @throws NoSuchEntityException
      */
-    public function getMerchantAccountWithClientkey(string $xapikey): array
+    public function getMerchantAccountWithApikey(string $xapikey, ?bool $demoMode = null): array
     {
         $storeId = $this->storeManager->getStore()->getId();
-        $client = $this->adyenHelper->initializeAdyenClient($storeId, $xapikey);
-        $management = new \Adyen\Service\Management($client);
+        $client = $this->adyenHelper->initializeAdyenClient($storeId, $xapikey, $demoMode);
+        $management = new Management($client);
         $response = $management->me->retrieve();
         return [
             'clientKey' => $response['clientKey'],
             'associatedMerchantAccounts' => $response['associatedMerchantAccounts'],
         ];
     }
+
+    /**
+     * @param string $apiKey
+     * @param string $merchantId
+     * @param string $username
+     * @param string $password
+     * @param string $url
+     * @param bool $demoMode
+     * @throws AdyenException
+     * @throws NoSuchEntityException
+     */
+    public function setupWebhookCredentials(
+        string $apiKey,
+        string $merchantId,
+        string $username,
+        string $password,
+        string $url,
+        bool $demoMode
+    ) {
+        $storeId = $this->storeManager->getStore()->getId();
+        $client = $this->adyenHelper->initializeAdyenClient($storeId, $apiKey, $demoMode);
+
+        $management = new Management($client);
+        $params = [
+            'url' => $url,
+            'username' => $username,
+            'password' => $password,
+            'communicationFormat' => 'json',
+            'active' => true,
+        ];
+        $webhookId = $this->configHelper->getWebhookId($storeId);
+        if ($webhookId) {
```

Comment:
Should we check for !empty here?
```suggestion
        if ($webhookId) {
```